libbeat/monitoring/report/elasticsearch/client.go

```diff
@@ -152,3 +148,88 @@ func (c *publishClient) Test(d testing.Driver) {
 func (c *publishClient) String() string {
 	return "publish(" + c.es.String() + ")"
 }
+
+func (c *publishClient) publishWithXPackMonitoringBulk(params map[string]string, event publisher.Event, t interface{}) error {
+	meta := common.MapStr{
+		"_index":   "",
+		"_routing": nil,
+	}
+	if c.es.GetVersion().Major < 7 {
+		meta["_type"] = t
+	}
+
+	action := common.MapStr{
+		"index": meta,
+	}
+
+	document := report.Event{
+		Timestamp: event.Content.Timestamp,
+		Fields:    event.Content.Fields,
+	}
+	bulk := [2]interface{}{action, document}
+
+	// Currently one request per event is sent. Reason is that each event can contain different
+	// interval params and X-Pack requires to send the interval param.
+	_, err := c.es.BulkWith("_xpack", "monitoring", params, nil, bulk[:])
+	return err
+}
+
+func (c *publishClient) publishWithBulk(event publisher.Event) error {
+	action := common.MapStr{
+		"index": common.MapStr{
+			"_index":   getMonitoringIndexName(),
+			"_routing": nil,
+		},
+	}
+
+	esVersion := c.es.GetVersion()
+	v7 := common.MustNewVersion("7.0.0")
+	if esVersion.LessThan(v7) {
+		action.Put("index._type", "doc")
+	}
```

Comment:
Create global const for v7 or check for esVersion.Major. Instead of `relying` on `action.Put` how about splitting it into action and meta, with `meta := common.MapStr{"_index", ..., "_routing"}`.